### PR TITLE
Refactor tabs system

### LINKED ISSUE #14: Refactor to initialize tab layout in separate class.

### PR DESCRIPTION
**Pull request by NoahMollerstuen**
Resolves #14

### CODE REVIEW

File: gui/app.py [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-    def __init__(self, cv_img, cam_index):
-        self.cv_img = cv_img
-        self.cam_index = cam_index
+@dataclasses.dataclass
+class Frame:
+    cv_img: np.ndarray
+    cam_index: int
+
+
+def convert_cv_qt(cv_img):
```
Comment (ryankarpuszka) [inline gui/app.py:16]: There were some changes to this function to support greyscale images you'll want to merge in here.

File: gui/app.py [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
 import cv2
 from PyQt5.QtCore import pyqtSignal, pyqtSlot, Qt, QThread
-import numpy as np
 
 
-class Frame():
-    def __init__(self, cv_img, cam_index):
-        self.cv_img = cv_img
-        self.cam_index = cam_index
+@dataclasses.dataclass
```
Comment (ryankarpuszka) [inline gui/app.py:10]: Fancy

**Review by ryankarpuszka — CHANGES_REQUESTED** [review]
Just resolve some merge conflicts and then this looks good to go.

**Comment by NoahMollerstuen** [comment]
I merged in the recent commits to main. These changes are gonna affect pretty much everything else we're working on, especially the VideoWidget, so ideally we should merge this in before too much more work gets done.

**Review by ryankarpuszka — APPROVED** [review]
This is a lot better organized now.

**Pull request merged**